mlflow/utils/rest_utils.py

```diff
@@ -14,26 +12,82 @@
 from mlflow.utils.string_utils import strip_suffix
 from mlflow.exceptions import MlflowException, RestException
 
-_REST_API_PATH_PREFIX = "/api/2.0"
 RESOURCE_DOES_NOT_EXIST = "RESOURCE_DOES_NOT_EXIST"
-
-_logger = logging.getLogger(__name__)
-
+_REST_API_PATH_PREFIX = "/api/2.0"
 _DEFAULT_HEADERS = {"User-Agent": "mlflow-python-client/%s" % __version__}
+# Response codes that generally indicate transient network failures and merit client retries,
+# based on guidance from cloud service providers
+# (https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines)
+_TRANSIENT_FAILURE_RESPONSE_CODES = frozenset(
+    [
+        408,  # Request Timeout
+        429,  # Too Many Requests
+        500,  # Internal Server Error
+        502,  # Bad Gateway
+        503,  # Service Unavailable
+        504,  # Gateway Timeout
+    ]
+)
+
+
+def _get_http_response_with_retries(
+    method, url, max_retries, backoff_factor, retry_codes, **kwargs
+):
+    """
+    Performs an HTTP request using Python's `requests` module with an automatic retry policy.
+
+    :method: a string indicating the method to use, e.g. "GET", "POST", "PUT".
+    :url: the target URL address for the HTTP request.
+    :param max_retries: Maximum total number of retries.
```

Comment:
Sorry I wasn't clear. Can you add `param` to other parameters?

Reply:
It's unclear what style I should follow from adjacent functions. Can you point me a canonical function example for style guide?

Reply:
Sure!

https://github.com/mlflow/mlflow/blob/8ef60e2dcac711a149fecac9e2b9ac2846610ad4/mlflow/tracking/fluent.py#L420-L430

Reply:
Thanks!